streams/src/main/java/org/apache/kafka/streams/processor/internals/ActiveTaskCreator.java

```diff
@@ -304,17 +291,11 @@ void closeAndRemoveTaskProducerIfNeeded(final TaskId id) {
     }
 
     Map<MetricName, Metric> producerMetrics() {
-        // When EOS is turned on, each task will have its own producer client
-        // and the producer object passed in here will be null. We would then iterate through
-        // all the active tasks and add their metrics to the output metrics map.
-        final Collection<StreamsProducer> producers = threadProducer != null ?
-            Collections.singleton(threadProducer) :
-            taskProducers.values();
-        return ClientUtils.producerMetrics(producers);
+        return ClientUtils.producerMetrics(Collections.singleton(threadProducer));
```

Comment:
We maybe could change `producerMetrics` to only take a single `Producer` instead of a `Collection`. However, it does not make a big difference, and we might add a second producer in the future, so maybe best to least as-is.